### PR TITLE
Odata feature matrix

### PR DESCRIPTION
**Pull request by artemkovalyov**
## Context

Long-awaited feature matrix. Don't merge, but add suggestions. It's work in progress

### CODE REVIEW

File: docs/java/features/odata/overview.md [hunk 1]
```diff
@@ -31,4 +38,51 @@ The Cloud SDK supports consumption of OData services through the following featu
 - A generator (available as command line interface and Maven plugin) that is capable of generating typed OData clients from service definitions (OData metadata documents)
 - Java Virtual Data Model (VDM): pre-generated typed OData client for [all OData services of SAP S/4HANA Cloud](https://api.sap.com/package/SAPS4HANACloud?section=Artifacts)
 
-The SDK supports both OData v2 and OData v4 protocols. However, parts of the v4 implementation are still in beta and not all features are supported yet.
+The SDK supports both OData v2 and OData v4 protocols.
+
+## OData features supported by SAP Cloud SDK for Java
```
Comment (MatKuhr) [inline docs/java/features/odata/overview.md:43]: In general I like the idea of a feature matrix. I would structure it a bit more to make it easier to find stuff. 

I could imagine for instance:

- Operations (CRUDQ Count Functions Actions Batch)
- Query Parameters (select, filter, top, skip etc.)
- Nested Queries
- Filter Functions
- Deep Operations (targeting Navigational Properties directly)

This also gives some context to the terms which are easily mistaken or confused as we know from our own expeirience.

Comment (MatKuhr) [reply docs/java/features/odata/overview.md:43]: A valuable addition might also be to add links for these sections to the documentation

**Comment by artemkovalyov** [comment]
@MatKuhr, @newtork, please, take another look. I'll need your help to thoroughly refine the Matrix. We can either push it in the current state and refine it with the Java team or let's have a meeting to do it together then.

To my opinion, it would already be Ok if it doesn't have false data and we can extend it iteratively before reaching completeness.

File: docs/java/features/overview.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (MatKuhr) [inline docs/java/features/overview.md:1]: I think this is a leftover?

File: docs/java/features/odata/overview.md [hunk 3]
```diff
@@ -16,19 +16,124 @@ keywords:
...
+- A [code generator](../generate-typed-odata-v2-and-v4-client-for-java) to convert your OData service definition into type-safe Java client library.
+  - Service definition usually comes in form of `.edmx` file and contains metadata of Odata service. The simplest way to obtain it if you have access to the service is navigating to `http(s)://<service-path>/$metadata` endpoint. You can easily generate a client for any 3rd party service or a service that you've build on you own.
+  - You can invoke code generator via our command line interface (CLI) or via Maven plugin
+- For many popular SAP OData services we ship pre-generated type-safe clients. In SAP universe they are called Virtual Data Model or VDM. The most popular type-safe client module contains a collection of [all OData services of SAP S/4HANA Cloud](https://api.sap.com/package/SAPS4HANACloud?section=Artifacts)
+- We support both OData v2 and OData v4 protocols in the SAP Cloud SDK for Java.
+
+## OData features supported by SAP Cloud SDK for Java
+
+### Create, Read, Update, Delete - CRUD
```
Comment (MatKuhr) [inline docs/java/features/odata/overview.md:46]: I like the improvement in structure. Looking at it again I would further improve by:

- Removing the Comment and Example query sections for now
- Linking the official OData 2.0 & 4.0 docu and
- Linking the Cloud SDK docu

I would do the last 2 things per section.

Also not all of the checkmarks are correct. Do you want to go over them together or should I just update the table?

Comment (artemkovalyov) [reply docs/java/features/odata/overview.md:46]: @MatKuhr 

This is exactly where I need help:

- Naming the features and agreeing on common names for them between Java & JS
- Putting correct checkmarks (I did it quick and dirty)
- I can have with linking of the Docu and official OData doc. That's also not where we have to do it in one go.

**Comment by artemkovalyov** [comment]
Closing it for now. It'll see another incarnation.

**Pull request closed without merging**